openedx/core/djangoapps/content/learning_sequences/api/outlines.py

```diff
@@ -490,7 +492,10 @@ def _update_sequences(course_outline: CourseOutlineData, course_context: CourseC
             LearningSequence.objects.update_or_create(
                 learning_context=course_context.learning_context,
                 usage_key=sequence_data.usage_key,
-                defaults={'title': sequence_data.title}
+                defaults={
+                    'usage_key_hash': hash_usage_key(sequence_data.usage_key),
+                    'title': sequence_data.title,
+                },
```

Comment:
Misc: You might find it easier to update tests and the like if default `usage_key_hash` generation happens as part of the `LearningSequenceData` attrs object.

Reply:
That's a good idea.

Reply:
FWIW, I'm not sure if the decorator form of defaults will let you access other attributes on initialization, e.g.

```python
@attr.s(frozen=True)
class LearningSequenceData:
    usage_key_hash = attr.ib()

    @usage_key_hash.default
    def _generate_usage_key_hash(self):
        return # hashed stuff
```

But in the worst case you can hook into the `__attrs_post_init__` hook like we do in `CourseOutlineData`.

Reply:
@ormsbee I incorporated this change in my latest commit -- `@usage_key_hash.default` works great!

Heads up that it involved pulling the contents of key_hashing.py into data.py, since we can't make `data` depend on `api`. If you'd like, I could split `key_hashing` it back into its own module, but it'd be at the root of `learning_sequences` instead of part of `learning_sequences.api`.

Reply:
I think it's fine to have it in `data.py` for now.

Reply:
Done.